pandas/core/dtypes/astype.py

```diff
@@ -258,6 +258,11 @@ def astype_is_view(dtype: DtypeObj, new_dtype: DtypeObj) -> bool:
     -------
     True if new data is a view or not guaranteed to be a copy, False otherwise
     """
+    if dtype.kind in "iufb" and dtype.kind == new_dtype.kind:
+        # fastpath for numeric dtypes
+        if hasattr(dtype, "itemsize") and hasattr(new_dtype, "itemsize"):
+            return dtype.itemsize == new_dtype.itemsize  # type: ignore[reportGeneralTypeIssues]
```

Comment:
When pyright finds an issue but not mypy, you have to use the pyright-specific ignore comment: `# pyright: ignore[report...]`